sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -349,6 +349,32 @@ statement
         (functionArgument (COMMA functionArgument)*)?
         RIGHT_PAREN                                                    #call
     | unsupportedHiveNativeCommands .*?                                #failNativeCommand
+    | createPipelineDatasetHeader (LEFT_PAREN colDefinitionList? RIGHT_PAREN)? tableProvider?
```

Comment:
shall we simply combine this with 
```
    | createTableHeader (LEFT_PAREN tableElementList RIGHT_PAREN)? tableProvider?
        createTableClauses
        (AS? query)?                                                   #createTable
```
?

Reply:
It depends on if the syntaxes will diverge further. e.g. `CREATE OR REFRESH MATERIALIZED VIEW`

Reply:
I also wonder if we should use `tableElementList` instead of `colDefinitionList` here. MV and ST should support table level constraints.